src/passes/InstrumentMemory.cpp

```diff
@@ -78,12 +78,16 @@ struct InstrumentMemory : public WalkerPass<PostWalker<InstrumentMemory>> {
   void visitLoad(Load* curr) {
     id++;
     Builder builder(*getModule());
+    auto ptrType = getModule()->memory.ptrType();
+    auto offset = getModule()->memory.is64
+                    ? builder.makeConst(curr->offset.addr)
+                    : builder.makeConst(int32_t(curr->offset.addr));
```

Comment:
is this the type of place where `makePtr` could help?

Reply:
That would be a different kind of `makePtr`. I'll add a utility method `makeConstPtr`.